src/mongroupdef.cpp

```diff
@@ -24,6 +24,11 @@ MonsterGroupResult MonsterGroupManager::GetResultFromGroup(
 {
     int spawn_chance = rng(1, 1000);
     MonsterGroup group = GetMonsterGroup( group_name );
+    int replace_time = DAYS(group.monster_group_time * ACTIVE_WORLD_OPTIONS["MONSTER_GROUP_DIFFICULTY"]) * (calendar::turn.season_length() / 14);
```

Comment:
Well, you mentioned "processing power standpoint", so I'm going to point to a major slowdown:

- `MonsterGroupManager::GetMonsterGroup` should return a **const reference** to the group, not a copy.
- You can store that reference as a pointer, if it should be replaced with another group, store the pointer to that other group there etc.
- no need to copy the monster group name into a temporary (only used once) variable

```C++
auto *groupptr = &GetMonsterGroup( ... );
while( groupptr->replace ) {
    groupptr = &GetMonsterGroup( groupptr->replace );
}
auto &group = *groupptr; // this only for compatibility with the following code
```
Even better: put that piece above into a function `GetReplacedMonsterGroup` (or whatever name you like) and let that function return a reference to the final group.